### PR TITLE
Feature/payment provider gateway klarnakp

### PR DESCRIPTION
**Pull request by LucianTuriacArnia**

### CODE REVIEW

**Comment by LucianTuriacArnia** [comment]
[BP-1968](https://jira.buckaroo.nl/browse/BP-1968)

File: Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        $totalOrder = $order->getBaseGrandTotal();
+        $this->numberOfInvoices = $order->getInvoiceCollection()->count();
+        $this->currentInvoiceTotal = 0;
+
+        // loop through invoices to get the last one (=current invoice)
+        if ($this->numberOfInvoices) {
+            $oInvoiceCollection = $order->getInvoiceCollection();
+
+            $i = 0;
```
Comment (harli91) [inline Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php:36]: what's the use of this `$i`?

File: Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        $order = $this->getOrder();
+
+        $totalOrder = $order->getBaseGrandTotal();
+        $this->numberOfInvoices = $order->getInvoiceCollection()->count();
+        $this->currentInvoiceTotal = 0;
+
+        // loop through invoices to get the last one (=current invoice)
+        if ($this->numberOfInvoices) {
+            $oInvoiceCollection = $order->getInvoiceCollection();
```
Comment (harli91) [inline Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php:34]: a collection usually has a `last()` method to get the last item in the collection, or use `end()`

File: Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php (hunk 1 above)
Comment (LucianTuriacArnia) [reply Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php:36]: $i is used to skip all invoice until the last one 

if (++$i !== $numberOfInvoices) {
                    continue;
}

The code was copied from app/code/Buckaroo/Magento2/Model/Method/AbstractMethod.php:1606

On Buckaroo/Magento2/Gateway/Request/InvoicedArticlesDataBuilder.php:37 I used filters to get the last one: 

$currentInvoice = $invoiceCollection
            ->addAttributeToSort('created_at', 'DSC')
            ->setPage(1, 1)
            ->getFirstItem();

Should I use the same for AbstractInvoiceDataBuilder?

File: Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php (hunk 2 above)
Comment (LucianTuriacArnia) [reply Gateway/Request/Invoice/AbstractInvoiceDataBuilder.php:34]: I used: $invoiceCollection->getLastItem() in both places

**Review by harli91 — APPROVED** [review]

**Pull request merged**